src/backend/common/jit/Node.hpp

```diff
@@ -228,6 +248,20 @@ class Node {
 
     /// Default destructor
     virtual ~Node() noexcept = default;
+
+    /// Returns the hash of the node. For all Nodes other than the Buffer node,
+    /// this is the pointer of the object
+    virtual size_t getHash() const noexcept {
+        size_t out = 0;
+        auto ptr   = this;
+        memcpy(&out, &ptr, std::max(sizeof(Node *), sizeof(size_t)));
+        return out;
```

Comment:
```suggestion
        constexpr size_t PtrSizeInBytes = sizeof(Node *) > sizeof(size_t) ? sizeof(Node *) : sizeof(size_t);
        size_t out = 0;
        std::memcpy(&out, &ptr, PtrSizeInBytes);
        return out;
```
This should help evaluate the copy size at compile time.